packages/tools/tests/test/audioV2/utils/audioEngineV2.utils.ts

```diff
@@ -0,0 +1,158 @@
+import type { Nullable } from "@dev/core/types";
+import { test } from "@playwright/test";
+import { getGlobalConfig } from "@tools/test-tools";
+
+/**
+ * The maximum pulse volume in the sound test file containing the pulse train.
+ */
+const MaxPulseVolume = 0.1;
+
+// Declarations for babylonServer/public/audiov2-test.js
+declare global {
+    let audioTestConfig: AudioTestConfig;
+    let audioTestResult: AudioTestResult;
+
+    class AudioV2Test {
+        public static AfterEachAsync(): Promise<void>;
+        public static BeforeEachAsync(): Promise<void>;
+        public static CreateAudioEngineAsync(options?: Partial<BABYLON.IWebAudioEngineOptions>): Promise<BABYLON.AudioEngineV2>;
+        public static CreateSoundAsync(source: string | string[], options?: Partial<BABYLON.IStaticSoundOptions>): Promise<any>;
+        public static GetResultAsync(): Promise<AudioTestResult>;
+        public static WaitAsync(seconds: number): Promise<void>;
+    }
+}
+
+test.beforeEach(async ({ page }) => {
+    await page.goto(getGlobalConfig().baseUrl + `/empty.html`, {
+        timeout: 0,
+    });
+
+    await page.waitForFunction(() => {
+        return window.BABYLON;
+    });
+
+    page.setDefaultTimeout(0);
+
+    await page.evaluate(
+        async ({ config }: { config: AudioTestConfig }) => {
+            audioTestConfig = config;
+
+            await BABYLON.Tools.LoadScriptAsync(audioTestConfig.baseUrl + "/audiov2-test.js");
+            await AudioV2Test.BeforeEachAsync();
+        },
+        { config: new AudioTestConfig() }
+    );
+});
+
+test.afterEach(async ({ page }) => {
+    await page.evaluate(async () => {
+        await AudioV2Test.AfterEachAsync();
+    });
+
```

Comment:
TODO: Save the audio results as audio files if the test failed.

Reply:
Where should audio file output be saved in this case? How is the path determined programmatically?

Reply:
Test results are stored in a specific directory - `packages/tools/tests/playwright-report`. If you save the audio files in this directory as well it will be uploaded to the snapshot server and will be available through the snapshot CDN.